assets/js/modules/search-console/util/site-stats-data.js

```diff
@@ -0,0 +1,108 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { getLocale, numFmt, calculateChange } from '../../../util';
+
+/**
+ * Gets data for a Google Chart from an search console report.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} current        A report object for the current period.
+ * @param {Object} previous       A report object for the previous period.
+ * @param {string} label          Metric label.
+ * @param {number} selectedColumn Selected column index.
+ * @return {Array.<Array.<number|string>>} Data array.
+ */
+export const getSiteStatsDataForGoogleChart = ( current, previous, label, selectedColumn ) => {
+	const dataMap = [
+		[
+			{ type: 'date', label: __( 'Day', 'google-site-kit' ) },
+			{ type: 'string', role: 'tooltip', p: { html: true } },
+			{ type: 'number', label },
+			{ type: 'number', label: __( 'Previous period', 'google-site-kit' ) },
+		],
+	];
+
+	const stringToDate = ( dateString ) => new Date( `${ dateString } 00:00:00` );
+	const locale = getLocale();
+	const localeDateOptions = {
+		weekday: 'short',
+		month: 'short',
+		day: 'numeric',
+	};
+
+	current.forEach( ( currentDay, index ) => {
+		const currentMonth = currentDay[ selectedColumn ];
+		const prevMonth = previous[ index ][ selectedColumn ];
+		const currentDate = currentDay.keys[ 0 ];
+		const previousDate = previous[ index ].keys[ 0 ];
+		const dateRange = sprintf(
+			/* translators: 1: date for user stats, 2: previous date for user stats comparison */
+			_x( '%1$s vs %2$s', 'Date range for chart tooltip', 'google-site-kit' ),
+			stringToDate( currentDate ).toLocaleDateString( locale, localeDateOptions ),
+			stringToDate( previousDate ).toLocaleDateString( locale, localeDateOptions ),
+		);
+		const change = calculateChange( prevMonth, currentMonth );
+		const difference = prevMonth !== 0
+			? ( currentMonth / prevMonth ) - 1
+			: 1; // if previous month has 0, we need to pretend it's 100% growth, thus the "difference" has to be 1
+
+		const statInfo = sprintf(
+			/* translators: 1: selected stat label, 2: numeric value of selected stat, 3: up or down arrow , 4: different change in percentage, %%: percent symbol */
+			_x( '%1$s: <strong>%2$s</strong> <em>%3$s %4$s%%</em>', 'Stat information for chart tooltip', 'google-site-kit' ),
+			label,
+			Math.abs( currentMonth ).toFixed( 2 ).replace( /(.00|0)$/, '' ),
+			`<svg width="9" height="9" viewBox="0 0 10 10" fill="none" xmlns="http://www.w3.org/2000/svg" class="${ classnames( 'googlesitekit-change-arrow', {
+				'googlesitekit-change-arrow--up': difference > 0,
+				'googlesitekit-change-arrow--down': difference < 0,
+			} ) }">
+				<path d="M5.625 10L5.625 2.375L9.125 5.875L10 5L5 -1.76555e-07L-2.7055e-07 5L0.875 5.875L4.375 2.375L4.375 10L5.625 10Z" fill="currentColor" />
+			</svg>`,
```

Comment:
Looks like we have 4 different functions in the codebase that use the same svg string like this. Let's move it into the global utility function to avoid repeating ourselves.

Reply:
`getChartDifferenceArrow` utility function added @eugene-manuilov